lib/portage/eapi.py

```diff
@@ -288,25 +414,39 @@ def eapi_has_sysroot(eapi):
 
 _eapi_attrs = collections.namedtuple(
     "_eapi_attrs",
-    "allows_package_provided "
-    "bdepend "
-    "broot "
-    "dots_in_PN dots_in_use_flags "
-    "exports_AA "
-    "exports_EBUILD_PHASE_FUNC "
-    "exports_ECLASSDIR "
-    "exports_KV "
-    "exports_merge_type "
-    "exports_PORTDIR "
-    "exports_replace_vars "
-    "feature_flag_test "
-    "idepend iuse_defaults iuse_effective posixish_locale "
-    "path_variables_end_with_trailing_slash "
-    "prefix "
-    "repo_deps required_use required_use_at_most_one_of "
-    "selective_src_uri_restriction slot_operator slot_deps "
-    "src_uri_arrows strong_blocks use_deps use_dep_defaults "
-    "empty_groups_always_true sysroot",
+    [
+        "allows_package_provided",
+        "bdepend",
+        "broot",
+        "dots_in_PN",
+        "dots_in_use_flags",
+        "exports_AA",
+        "exports_EBUILD_PHASE_FUNC",
+        "exports_ECLASSDIR",
+        "exports_KV",
+        "exports_merge_type",
+        "exports_PORTDIR",
+        "exports_replace_vars",
+        "feature_flag_test",
+        "idepend",
+        "iuse_defaults",
+        "iuse_effective",
+        "posixish_locale",
+        "path_variables_end_with_trailing_slash",
+        "prefix",
+        "repo_deps",
+        "required_use",
+        "required_use_at_most_one_of",
+        "selective_src_uri_restriction",
+        "slot_operator",
+        "slot_deps",
+        "src_uri_arrows",
+        "strong_blocks",
+        "use_deps",
+        "use_dep_defaults",
+        "empty_groups_always_true",
+        "sysroot",
+    ],
```

Comment:
I don't understand the intention of this change. Is there reason to believe that a list of field names is somehow better than a space-separated list of field names?

Reply:
A real tuple or list is certainly more Pythonic than whitespace-delimited-string fake list.
(If tuple instead of list is working, then I suggest to use tuple, i.e. change `[` `]` to `(` `)` around those elements.)

Reply:
`tuple` works fine, I'll make that change.

@mattst88 from a technical standpoint, either the list or the space-separated string work equally fine.

However, it's my opinion that the space-separated list is more difficult to reason about, as well as more error prone.

For example, in the original code, [line 306](https://github.com/gentoo/portage/blob/master/lib/portage/eapi.py#L306) identifies three named fields. This is easy to gloss over, though, as the majority of the fields are defined one-per-line.

Additionally, using this style lends itself to errors such as:

```py
  "my_new_field",
  "some_other_field ",
```

where a missing space results in only _one_ additional field being added versus the expected two.